test/uk/gov/hmrc/hec/controllers/IFControllerSpec.scala

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.hec.controllers
+
+import java.time.LocalDate
+
+import cats.data.EitherT
+import cats.instances.future._
+import play.api.inject.bind
+import play.api.inject.guice.GuiceableModule
+import play.api.test.FakeRequest
+import play.api.test.Helpers._
+import uk.gov.hmrc.hec.models.ids.{CTUTR, SAUTR}
+import uk.gov.hmrc.hec.models.{CTStatus, CTStatusResponse, Error, SAStatus, SAStatusResponse, TaxYear}
+import uk.gov.hmrc.hec.services.IFService
+import uk.gov.hmrc.http.HeaderCarrier
+
+import scala.concurrent.ExecutionContext.Implicits.global
+
+class IFControllerSpec extends ControllerSpec {
+
+  val mockIFService = mock[IFService]
+
+  override val overrideBindings =
+    List[GuiceableModule](
+      bind[IFService].toInstance(mockIFService)
+    )
+
+  val controller = instanceOf[IFController]
+
+  def mockGetSAStatus(utr: SAUTR, taxYear: TaxYear)(result: Either[Error, SAStatusResponse]) =
+    (mockIFService
+      .getSAStatus(_: SAUTR, _: TaxYear)(_: HeaderCarrier))
+      .expects(utr, taxYear, *)
+      .returning(EitherT.fromEither(result))
+
+  def mockGetCTStatus(utr: CTUTR, fromDate: LocalDate, toDate: LocalDate)(result: Either[Error, CTStatusResponse]) =
+    (mockIFService
+      .getCTStatus(_: CTUTR, _: LocalDate, _: LocalDate)(_: HeaderCarrier))
+      .expects(utr, fromDate, toDate, *)
+      .returning(EitherT.fromEither(result))
+
+  "IFController" when {
+
+    "handling requests to fetch SA status" must {
+
+      val validSautr = "1000062440"
+
+      "return a 400 (bad request)" when {
+
+        def testBadRequest(utr: String, taxYear: String) = {
+          val call    = routes.IFController.getSAStatus(utr, taxYear)
+          val request = FakeRequest(call)
```

Comment:
the way the tests work, I think you can get away with having just `FakeRequest()` without the correct call in it as long as we call the correct method on `controller` here

Reply:
true! updated

Reply:
did you push up the update? isn't this the same things as before?

Reply:
looks like I've missed some, sorry